test/test_utils/__init__.py

```diff
@@ -554,3 +554,20 @@ def get_repository_and_tag_from_image_uri(image_uri):
     repository_uri, tag = image_uri.split(":")
     _, repository_name = repository_uri.split("/")
     return repository_name, tag
+
+
+def get_processor_from_image_uri(image_uri):
+    """
+    Return processor from the image URI
+
+    Assumes image uri includes -<processor>- in it's tag, where <processor> is one of cpu, gpu or eia.
+
+    :param image_uri: ECR image URI
+    :return: cpu, gpu, or eia
+    """
+    allowed_processors = ("cpu", "gpu", "eia")
+
+    for processor in allowed_processors:
+        match = re.search(rf'-({processor})-', image_uri)
```

Comment:
Please remove the second `-` in this line, and make it `rf"-({processor})"`, so that this function can be used on TensorFlow Inference image tags such as `tensorflow-inference:2.2.0-cpu`

Reply:
Many of our other tests are not explicitly used for SM tags, but will add this for now.

Reply:
Thanks. This function is usable across a wide range of tests, and having it be generic to test on all our DLC image tags will be better than locking our code to be specifically usable only for CI builds.